src/model/test/ModelMerger_GTest.cpp

```diff
@@ -869,3 +871,36 @@ TEST_F(ModelFixture, ModelMerger_SuggestMapping_ExampleModel) {
     EXPECT_EQ(model1.getObject(mapPair.first)->nameString(), model2.getObject(mapPair.second)->nameString());
   }
 }
+
+TEST_F(ModelFixture, ModelMerger_ClimateZones_4166) {
+
+  // Test for #4166 - Merging FloorSpaceJS strips out climate zone assignment.
+
+  Model model1;
+  Model model2;
+
+  // Add a Climate Zone to model1 only
+  ClimateZones czs = model1.getUniqueModelObject<ClimateZones>();
+  ClimateZone cz = czs.setClimateZone(ClimateZones::ashraeInstitutionName(), "4A");
+  EXPECT_EQ(1, czs.numClimateZones());
+  EXPECT_EQ(1, czs.climateZones().size());
+
+  // To reproduce the original issue, we also need a Site object since it's the fact that the Site object is deleted, and along with it its children
+  // and that includes the ClimateZones
+  Site site1 = model1.getUniqueModelObject<Site>();
+  ASSERT_EQ(1, site1.children().size());
+  EXPECT_EQ(czs, site1.children().front());
+
+  EXPECT_TRUE(model1.getOptionalUniqueModelObject<ClimateZones>());
+  EXPECT_FALSE(model2.getOptionalUniqueModelObject<ClimateZones>());
+
+  // do merge
+  ModelMerger mm;
+  std::map<UUID, UUID> handleMapping;
+  mm.mergeModels(model1, model2, handleMapping);
+
+  ASSERT_TRUE(model1.getOptionalUniqueModelObject<ClimateZones>());
```

Comment:
This line fails before fix. This is the MCVE that exhibits the issue present in #4166